resources/[system]/scoreboard/server.lua

```diff
@@ -0,0 +1,101 @@
+RegisterNetEvent("scoreboard:getPlayers")
+AddEventHandler("scoreboard:getPlayers", function()
+  TriggerClientEvent("scoreboard:receivePlayers", source, players)
+end)
+
+columns = {
+  {
+    friendlyName = "Player ID",
+    defaultValue = 0,
+    position = 0
+  },
+  {
+    friendlyName = "Name",
+    defaultValue = "***Invalid***",
+    position = 1
+  }
+}
+
+players = {}
+
+RegisterNetEvent("scoreboard:requestColumns")
+AddEventHandler("scoreboard:requestColumns", function()
+  players[source] = {}
+  for id, columnData in pairs(columns) do
+    players[source][columnData.friendlyName] = columnData.defaultValue
+  end
+  players[source]["Player ID"] = source
+  players[source]["Name"] = GetPlayerName(source)
+  TriggerClientEvent("scoreboard:receiveColumns", source, columns)
+end)
+
+--[[
+Event name: scoreboard:addColumn
+Description: Adds a column to the scoreboard in the specified position with a default value.
+Parameters:
+_friendlyName: Name of the column you want to add (shows on scoreboard).
+_defaultValue: Default value of the column.
+_position: The position of the column.
+]]
+AddEventHandler("scoreboard:addColumn", function(_friendlyName, _defaultValue, _position)
+  columns[#columns+1] = {
+    friendlyName = _friendlyName,
+    defaultValue = _defaultValue or 0,
+    position = _position
+  }
+  for _, playerId in pairs(GetPlayers()) do
+    if players[tonumber(playerId)] ~= nil then
+      players[tonumber(playerId)][_friendlyName] = _defaultValue
+    end
+  end
+  TriggerClientEvent("scoreboard:receiveColumns", -1, columns)
+end)
+
+--[[
+Event name: scoreboard:removeColumn
+Description: Removes the specified column.
+Parameters:
+columnName: Name of the column you want to remove (needs to be the same as the column name on the scoreboard).
+]]
+AddEventHandler("scoreboard:removeColumn", function(columnName)
+  local columnId = getColumnFromName(columnName)
+  if columnId then
+    for _, playerId in pairs(GetPlayers()) do
+      if players[tonumber(playerId)] ~= nil then
+        players[tonumber(playerId)][columnName] = nil
+      end
+    end
+    columns[columnId] = nil
+    TriggerClientEvent("scoreboard:receiveColumns", -1, columns)
+    TriggerClientEvent("scoreboard:receivePlayers", -1, players)
+  end
+end)
+
+--[[
+Event name: scoreboard:updateColumnValue
+Description: Updates the specified column value for the specified player.
+Parameters:
+src: The source of the player you want to change.
+columnName: Name of the column you want to change (needs to be the same as the column name on the scoreboard).
+value: The value of the column you want to set for that player.
+]]
+AddEventHandler("scoreboard:updateColumnValue", function(src, columnName, value)
+  local columnId = getColumnFromName(columnName)
+  if columnId then
+    players[src][columnName] = value
```

Comment:
Should just be state bags and a column should contain a key to be found in a state bag.

This'd also allow name/such columns to be some sort of meta if you'd, say, have 'default value' be replaced with a placeholder in such a case (and will allow stuff like RP names more trivially).

Reply:
We tried to use state bags at the start (hence my question in Matrix earlier today) however quickly realised that I couldn't get other players state bags when they are out of scope, do you have an example on how you would do this using state bags?